runtime/j9vm/jvm.c

```diff
@@ -1995,6 +2023,12 @@ jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
 		f_threadDetach(attachedThread);
 	}
 #endif /* CALL_BUNDLED_FUNCTIONS_DIRECTLY */
+#ifdef WIN32
+	if (NULL != executableJarPath) {
+		PORT_ACCESS_FROM_PORT(&j9portLibrary);
+		j9mem_free_memory(executableJarPath);
```

Comment:
The above code assigns this to a longer lived structure.
```c
specialArgs.executableJarPath = executableJarPath;
```

Why is it being freed here rather than when the longer lived structure is freed?

Reply:
`specialArgs.executableJarPath` is only used by https://github.com/eclipse/openj9/blob/5a3774c9e698f1e0661420efebc0d93dd6075d1e/runtime/j9vm/jvm.c#L1845
which uses the path to load the jar files and doesn't require the path afterwards.
https://github.com/eclipse/openj9/blob/5a3774c9e698f1e0661420efebc0d93dd6075d1e/runtime/util/vmargs.c#L1160-L1168
So `executableJarPath` can be freed after that usage while `J9VMInitArgs *vmArgumentsList` can be freed later.

Reply:
Can you add a comment here as well about why this is OK?

Reply:
Updated with a comment.